### PR TITLE
RIA-6532 - Adding flag to make submit hearing requirements not re-sub…

### PR DESCRIPTION
**Pull request by Dbarnes-v1**
## JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-6532

### Change description ###
Adding flag to make submit hearing requirements not re-submittable for ADA cases

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/ReviewHearingRequirementsConfirmation.java [hunk 1]
```diff
@@ -35,11 +37,25 @@ public PostSubmitCallbackResponse handle(
         PostSubmitCallbackResponse postSubmitResponse =
             new PostSubmitCallbackResponse();
 
+        final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        boolean isAcceleratedDetainedAppeal = asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class)
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/ReviewHearingRequirementsConfirmation.java:42]: can you check if there is an existing helper method to check if it is an ADA type in HandlerUtils. If not, please add the method there.

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/ReviewHearingRequirementsConfirmation.java (hunk 1 above)
Comment (Dbarnes-v1) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/ReviewHearingRequirementsConfirmation.java]: Added in the HandlerUtil & updated Boolean checks to use that in the handlers

**Review by colmlynch60 — DISMISSED** [review]
LGTM

**Review by Marcellofabbri — APPROVED** [review]
Tried locally, seems fine

**Review by colmlynch60 — APPROVED** [review]
LGTM

**Review by Marcellofabbri — APPROVED** [review]

**Pull request merged**